src/AbstractOperations/conditional_operations.jl

```diff
@@ -82,10 +85,21 @@ function ConditionalOperation(operand::AbstractField;
                               condition = nothing,
                               mask = zero(eltype(operand)))
 
+    condition = validate_condition(condition, operand)
+
     LX, LY, LZ = location(operand)
     return ConditionalOperation{LX, LY, LZ}(operand, func, operand.grid, condition, mask)
 end
 
+validate_condition(cond, ::AbstractField) = cond # fallback
+
+function validate_condition(cond::AbstractArray, operand::AbstractField)
+    if ndims(cond) != 3
```

Comment:
Should we instead check if 

```size(cond) == size(operand)```?

`operand` will always have three dimensions so the above should work for both the 3D check but also the size?

Reply:
Especially as your Argument error below says the condition must be the size of the operand

Reply:
hm... I think that's a good point
@glwagner?

Reply:
makes sense to me!